MIGRATION.md

```diff
@@ -0,0 +1,90 @@
+# Migrating to the `cloudant-java-sdk` library
+This document is to assist in migrating from the `java-cloudant` (package: `com.cloudant.cloudant-client`) to the newly supported [`cloudant-java-sdk`](https://github.com/IBM/cloudant-java-sdk) (package: `com.ibm.cloud.cloudant`).
+
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-java-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-java-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-java-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-java-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-java-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. Fetching the Database object first before performing additional operations is not required. For example, in the case of updating a document you would first call `getDocument` to fetch and then `putDocument` to update.
+2. Model classes are used instead of POJOs.
+3. Sending and receiving plain objects and streaming are both available. E.g. the `db.find` equivalent `getDocument` is returning with a `Document` object, meanwhile `getDocumentAsStream` is returning with an `InputStream`.
```

Comment:
How about adding links to the `...AsStream` examples in the API docs?

Reply:
Maybe we should add the `...AsStream` operations to the reference table and then mention those operations here.

Reply:
I have added them to the table in cc74fbf. I used the original link since there is no hook that can jump on the `AsStream` examples directly, and I excluded the ones out from the links where there is no example for the operation as a stream.